packages/calypso-build/webpack/minify.js

```diff
@@ -42,14 +42,23 @@ function chooseTerserEcmaVersion( browsers ) {
  */
 module.exports = ( options ) => {
 	let terserOptions = options.terserOptions || {};
+	const isDesktop = options.isDesktop || false;
```

Comment:
No need to leak the `isDesktop` abstraction to this file at all, it already provides a mechanism to overwrite all these options.

In `webpack.config.js`, try this:

```
minimizer: Minify( {
    //...
    terserOptions: {
        ...( isDesktop
            ? {
                    mangle: true,
                    ecma: 2017,
                    compress: false,
                    safari10: false,
                }
            : {
                    mangle: true,
                        compress: true
                } ),
    },
} ),
```

It should generate the config for Terser:

```
// For desktop
terserOptions: {
    ecma: 2017,
    ie8: false,
    safari10: false,
    mangle: true,
    compress: false
}

//For browser (evergreen)
terserOptions: { ecma: 2017, ie8: false, safari10: false, mangle: true }

//For browser (fallback)
terserOptions: { ecma: 5, ie8: false, safari10: false, mangle: true }
```

Reply:
🤔  I took a stab at doing it this way, but not sure if possible. The problem is the way `minify.js` is currently set up:

```
module.exports = ( options ) => {
	let terserOptions = options.terserOptions || {};
	terserOptions = {
		ecma: chooseTerserEcmaVersion( supportedBrowsers ),
		ie8: false,
		safari10: supportedBrowsers.some(
			( browser ) => browser.includes( 'safari 10' ) || browser.includes( 'ios_saf 10' )
		),
		...terserOptions,
	};

	return [ new TerserPlugin( { ...options, terserOptions } ) ];
};
```

`terserOptions` from `webpack.config.js` are simply spread here. Because we're not using something like `Object.assign`, these properties (safari, ie, etc) won't actually be overriden accordingly. With how the file is set up currently, it doesn't seem there is a way to make desktop-specific overrides without injecting an `isDesktop` flag into the config somehow.

Reply:
I tried refactoring the webpack overrides in e138309bd4a96ab1aa92cef2d478a9c1713a8e3d - let me know what you think.

Reply:
> Because we're not using something like Object.assign, these properties (safari, ie, etc) won't actually be overridden accordingly.

They are overridden by `...terserOptions`. If `options.terserOptions` is `{safari10: false}`, the above is equivalent to `terserOptions = { safari10: supportedBrowsers.some(...), ...{safari10:false}}` which evaluates to `terserOptions={safari10:false}`

I agree we can't override `browsers` directly, but we could just pass `ecma: 2017` which would the the result of `ecma: chooseTerserEcmaVersion('last 2 Chrome versions')`

Reply:
Cool beans. My JS-foo isn't very good so the end-result of the way the file was set up wasn't immediately obvious. Fixed in 126e8871e7f25e8ae4df260442495705caeb7094.

Reply:
It is not obvious at all :) I had to add a few `console.log()` and do a bunch of experiments before fully understanding what's going on